src/CachedCalls.jl

```diff
@@ -78,15 +89,37 @@ end
 """
     _deconstruct(ex)
 
-Deconstruct expression `ex` to a tuple of function name, arguments, and keyword arguments.
+Deconstruct expression `ex` to a tuple (function name, arguments, kwarg names, kwarg values)
 """
 function _deconstruct(ex)
-    func, fargs = Iterators.peel(ex.args)
-    length(ex.args) == 1 && return func, [], []
+    # escaped expressions need their unescaped subexpression deconstructed
+    isesc = Meta.isexpr(ex, :escape)
+    if isesc
+        nonesc_ex = ex.args[1]
+    else
+        nonesc_ex = ex
+    end
```

Comment:
This is just style but maybe a good use of ternary operator here?

```suggestion
    nonesc_ex = isesc ? ex.args[1] : ex
```